### DISCUSSION TITLE
How to recreate a realistic cymbal swell in Hydrogen?

### DISCUSSION THREAD

**Discussion by RockVicedomini** [hydrogen-music/hydrogen#1952] [Q&A] [answered]
Hello,
I am wondering if there is an easy way to generate a more realistic cymbal swell with hydrogen.
Just modulating notes velocity is generally not enough because the sound of the tip hitting the cymbal is noticeably louder than the level of the sound that we want to recreate.   
I am thinking to increase the attack time of the samples only if the software detects multiple hits on the same cymbal. 

Do you think this feature can be implemented?

**Comment by theGreatWhiteShark** [comment]
Hey @RockVicedomini ,

Have you tried a kit using several layers/samples for these cymbals? There notes do not just scale a sample's volume but also pick different samples based on their velocity. Especially for use cases like yours this gives a far more natural sound.

**Reply by RockVicedomini** [reply]
Yes, I am already using a multilayer kit (AVLDrumkit-BlondeBop), but I am able to produce a nice and smoot swell only tweaking the attack time from 0.00 (single crash hit) to something in the  range of 0.25 - 0.40 (multiple soft hits).
Unfortunately I am not able to set the attack time to a different value for each layer, so I cannot reproduce the effect I want in real time, adjusting the velocity.

**Reply by theGreatWhiteShark** [reply, marked as answer]
> Unfortunately I am not able to set the attack time to a different value for each layer

The attack value specifies the time (I think in milliseconds) during which the volume of the sample is multiplied with a value between 0 and 1. (See #1469 for the shape of the ADSR curve).

And it acts on the whole instrument. Right. Mostly because it is an ancient feature and probably predates multi-layered instruments in Hydrogen.

But you could another feature: the Sample Editor. Within the "Layer" tab of the Instrument Editor select a layer and hit the "Edit Layer" button. In the dialog that opens you can apply a custom envelope to the sample. Clicking the widget creates a new node, which can be dragged later on. (be sure to have the bottom right combo box set to "volume"). By approximately redrawing the ADSR envelope for each layer you can set custom attacks without altering the original samples.

**Reply by RockVicedomini** [reply]
Great, the sample editor makes the magic... 
I was not aware of the envelope editor feature. I will play with it a little bit to find the sweetspot I need to make it sound perfectly smooth. 

Many thanks for your advice.

**Reply by theGreatWhiteShark** [reply]
> I was not aware of the envelope editor feature.

Well, the dialog does a pretty bad job advertising the feature. It's one of those more esoteric features you stumble upon when writing the docs or take a look at the code itself.

A word of warning: the Sample Editor is still quite rough around the edges. (I did not had time to look at it at all). One particular thing to keep in mind is that all those custom envelop points are stored as part of the .h2song. Loading a different drumkit will discard them and saving the drumkit will _not_ save the envelop points as part of it. (At least for now. Starting with Hydrogen 1.3.0 you can persistently store them in .h2drumkit files).

**Reply by RockVicedomini** [reply]
Looking forward to see how many hidden gems still reserve this wonderful software. 
Thank you for your work.
